nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,306 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  cfg = config.services.draupnir;
+
+  format = pkgs.formats.yaml {};
+  configFile = format.generate "draupnir.yaml" cfg.settings;
+in
+{
+  #region Options
+  options.services.draupnir = {
+    enable = lib.mkEnableOption ("Draupnir, a moderation tool for Matrix");
+
+    package = lib.mkPackageOption pkgs "draupnir" { };
+
+    accessTokenFile = lib.mkOption {
+      type = with lib.types; nullOr path;
+      default = null;
+      description = ''
+        File containing the access token for Draupnir's Matrix account.
+        Make sure this does not contain newlines if writing manually: `:set nofixeol` for vim or -L for nano.
```

Comment:
```suggestion
        Make sure this does not contain newlines if writing manually: `:set noeol nofixeol` for vim or -L for nano.
```

Reply:
I'm guessing this trims a newline if one exists? Wouldn't that be negated by `nofixeol`?

Reply:
I had to add combine both options to afterwards save the file without a newline.
Quoting `:h noeol`:
>When writing a file and this option is off and [...] 'fixeol' option is off, no `<EOL>` will be written for the line in the file.